### PR TITLE
[ISSUE-586] Change volume usage after flipping disk from BAD to GOOD

### LINKED ISSUE #586: Volume should be transited to IN_USE state,  if drive health has been changed from BAD to GOOD
Volume is in FAILED state, if drive health has been changed of a drive from BAD to GOOD.

### PR DESCRIPTION
**Pull request by AnnaDutova**
Signed-off-by: Dutova <Ann.Dutova@Dell.com>

### Resolves #586 

Add steps for changing volumes usage after `action=add`

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [x] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
Add comment to issue how to reproduce this task 
Final result:
```
ubuntu:/Workspace/csi-baremetal/tests/app # kubectl get drive -o wide 
NAME                                   SIZE        TYPE   HEALTH   STATUS   USAGE    SYSTEM   PATH          SERIAL NUMBER        NODE                                   SLOT
06a40082-a3d1-47a8-bf3c-d2e8ad63454e   105906176   HDD    GOOD     ONLINE   IN_USE            /dev/loop15   LOOPBACK3369231940   d080dee4-aef8-40f0-8553-708626fecc1a   
46a84143-83ef-41da-9a14-30a93ab95915   105906176   HDD    GOOD     ONLINE   IN_USE            /dev/loop13   LOOPBACK2231103841   d080dee4-aef8-40f0-8553-708626fecc1a   
<. . .>
ubuntu:/Workspace/csi-baremetal/tests/app # kubectl get volume -o wide | grep 06a40082-a3d1-47a8-bf3c-d2e8ad63454e
pvc-8cef8e24-1cd9-46d6-a8d5-d7b2f3ac3e62   105906176   HDD             GOOD     PUBLISHED    OPERATIVE   IN_USE   xfs    06a40082-a3d1-47a8-bf3c-d2e8ad63454e   d080dee4-aef8-40f0-8553-708626fecc1a

```
pass custom-acceptance - 493

### CODE REVIEW

File: pkg/crcontrollers/drive/drivecontroller.go [hunk 1]
```diff
@@ -215,6 +215,22 @@ func (c *Controller) handleDriveUpdate(ctx context.Context, log *logrus.Entry, d
...
 		if c.checkAndPlaceStatusInUse(drive) {
+			volumes, err := c.crHelper.GetVolumesByLocation(ctx, id)
+			if err != nil {
+				return ignore, err
+			}
+			for _, vol := range volumes {
+				value, found := vol.GetAnnotations()[apiV1.VolumeAnnotationRelease]
+				if found && value == apiV1.VolumeAnnotationReleaseFailed {
+					vol.Spec.Usage = apiV1.VolumeUsageInUse
+					delete(vol.Annotations, apiV1.VolumeAnnotationRelease)
```
Comment (rkiselenko) [inline pkg/crcontrollers/drive/drivecontroller.go:226]: There is a special method `SetAnnotations(annotations)`.

File: pkg/crcontrollers/drive/drivecontroller.go [hunk 2]
```diff
@@ -215,6 +215,22 @@ func (c *Controller) handleDriveUpdate(ctx context.Context, log *logrus.Entry, d
...
+					if err := c.client.UpdateCR(ctx, vol); err != nil {
+						log.Errorf("Unable to change volume %s usage status to %s, error: %v.",
```
Comment (rkiselenko) [inline pkg/crcontrollers/drive/drivecontroller.go:228]: Can we reflect old and new status in the log entry?

**Review by rkiselenko — CHANGES_REQUESTED** [review]

File: pkg/crcontrollers/drive/drivecontroller.go [hunk 3]
```diff
@@ -215,6 +215,22 @@ func (c *Controller) handleDriveUpdate(ctx context.Context, log *logrus.Entry, d
...
+							vol.Name, vol.Spec.Usage, err)
+						return ignore, err
```
Comment (mishoyama) [inline pkg/crcontrollers/drive/drivecontroller.go:230]: Will we retry if update failed (by timeout, for example)?

Comment (AnnaDutova) [reply pkg/crcontrollers/drive/drivecontroller.go:230]: Hmm, if it failed, we [will return](https://github.com/dell/csi-baremetal/blob/d9a2e40a36b06d5857794139e4596f843afc227d/pkg/crcontrollers/drive/drivecontroller.go#L121) an error and don't update anything(disk/volume), then the parent's method(which called `Reconcile`) will get an unsatisfactory response and it makes more sense that **it** will try to send an update request again and again until the error disappears or the time runs out.

File: pkg/crcontrollers/drive/drivecontroller.go [hunk 4]
```diff
@@ -239,6 +243,29 @@ func getDriveAnnotationRemoval(annotations map[string]string) (string, bool) {
 	return status, found
 }
 
+func (c *Controller) changeVolumeUsageAfterActionAnnotation(ctx context.Context, log *logrus.Entry, drive *drivecrd.Drive) error {
```
Comment (mishoyama) [inline pkg/crcontrollers/drive/drivecontroller.go:246]: Please add/update UTs for this method.

**Review by rkiselenko — APPROVED** [review]

File: pkg/crcontrollers/drive/drivecontroller_test.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+	if err != nil {
+		panic(err)
+	}
+	return kubeClient
+}
+
+
+func TestDriveController_NewDriveController(t *testing.T){
+	kubeClient := setup()
+	dc := NewController(kubeClient, nodeID, nil, new(events.Recorder), testLogger)
```
Comment (safronovD) [inline pkg/crcontrollers/drive/drivecontroller_test.go:74]: It's better to use mock for recorder even if it has no calls. For example `mocks.NoOpRecorder`

**Review by mishoyama — APPROVED** [review]

File: pkg/crcontrollers/drive/drivecontroller_test.go (hunk 5 above)
Comment (AnnaDutova) [reply pkg/crcontrollers/drive/drivecontroller_test.go:75]: For using `mocks` need to have interface `eventRecorder` in drivecontroller

**Review by yurkov-anton — APPROVED** [review]

**Review by safronovD — APPROVED** [review]

**Review by yurkov-anton — APPROVED** [review]

**Pull request merged**